### PR TITLE
Fix cas properties declaration in JSP.

### PR DESCRIPTION
**Pull request by RemiDesgrange**
The new version is less readable than  `${myVar}` but much better than
the solution in 18.06 version.

### CODE REVIEW

**Comment by landryb** [comment]
If this is for #2569, don't forget that there are other instances of `${instanceName}` and that `${consoleContextPath}` & `${publicUrl}` also needs handling. Will test your PR .. thanks!

**Comment by landryb** [comment]
Fwiw, that seems to work here at runtime, tested with `headerHeight` & `headerUrl` defined in `cas.properties`, then commented out & only defined in `defaults.properties`, the values seems properly propagated - `instanceName` too.

Minor nit but for cleanlyness, the reindenting done in the first chunk of `propertyFileConfigurer.xml` is unrelated/unneeded.

So the principle looks right, so if you can also fix `consoleContextpath` in https://github.com/georchestra/georchestra/blob/19.04/cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginView.jsp#L65, `publicUrl` in https://github.com/georchestra/georchestra/blob/19.04/cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casGenericSuccess.jsp#L24 and `instanceName` in https://github.com/georchestra/georchestra/blob/19.04/cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/includes/top-noheader.jsp#L35 (hope i havent forgotten any) i think that'll be all green for me to merge

**Comment by RemiDesgrange** [comment]
I found a clearer solution. testing it and pushing asap

**Comment by landryb** [comment]
That indeed looks clearer, nice !

File: cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginView.jsp [hunk 1]
```diff
@@ -20,6 +20,10 @@
 --%>
 <jsp:directive.include file="includes/top.jsp" />
 
+<spring:eval expression="@propertyConfigurer.getProperty('consoleContextpath')" var="consoleContextpath"/>
+<spring:eval expression="@propertyConfigurer.getProperty('loginTicket')" var="loginTicket"/>
+<spring:eval expression="@propertyConfigurer.getProperty('flowExecutionKey')" var="flowExecutionKey"/>
+
```
Comment (landryb) [inline cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginView.jsp]: Are you sure loginTicket & flowExecutionKey need special treatment here ? to me only the vars from https://github.com/georchestra/georchestra/blob/19.04/cas-server-webapp/src/main/webapp/WEB-INF/cas-servlet.xml#L70 do...

**Comment by landryb** [comment]
Something feels wrong about `${consoleContextPath}` replacement, here i dont define a value in default.properties nor cas/cas.properties (where its defined as `cas/cas.properties:#console.contextpath=/console`), there's a default value in `cas-servlet.xml`
```
<entry key="consoleContextPath" value="${console.contextpath:/console}" />
```
but adding
```
<spring:eval expression="@propertyConfigurer.getProperty('consoleContextpath')" var="consoleContextpath"/>
```
to `casLoginView.jsp` doesnt seem enough, there's still an empty value for `consoleContextpath` and the two links displayed on the cas login page aren't prefixed by `/console` as expected by the defaults.

**Comment by landryb** [comment]
Uncommenting the default value in `cas/cas.properties` isnt enough to fix it, one needs to change the `casLoginView.jsp` line for:
```
<spring:eval expression="@propertyConfigurer.getProperty('console.contextpath')" var="consoleContextpath"/>
``` (note the right property name)

so imo, if we go that route the confusing default from `cas-servlet.xml` should be dropped, and the value should be uncommented from `cas.properties` so that it's properly set.

**Comment by RemiDesgrange** [comment]
Totally agree about `contextpath`

**Comment by RemiDesgrange** [comment]
Ok @landryb I fixed it for 19.04. but the whole config loading in cas need to be rethink, there's stuff everywhere.

**Comment by landryb** [comment]
Tested it here, seems to work fine (commented out values set in `cas.properties`, so `headerHeight/instanceName/headerUrl` properly come from `defaults.properties`), but it'd be better if someone else also tests it at runtime :) @fvanderbiest ?.

I wouldnt add the `loginTicket/flowExecutionKey` vars though...

**Comment by RemiDesgrange** [comment]
WDYT, ATM, this PR seems really fragile... I'm working on another solution but I'm fighting against spring, and, for now, it wins :-/

**Comment by groldan** [comment]
I think instead of doing `expression="@propertyConfigurer.getProperty(...)`  you can do  `expression="@environment.getProperty(...)` so it gets whatever value was already resolved by spring, being it a default or provided value

**Comment by landryb** [comment]
> I think instead of doing `expression="@propertyConfigurer.getProperty(...)` you can do `expression="@environment.getProperty(...)` so it gets whatever value was already resolved by spring, being it a default or provided value

which means there's no need for the `context:property-placeholder` replacement by a `propertyfactory`?

**Comment by RemiDesgrange** [comment]
> which means there's no need for the `context:property-placeholder` replacement by a `propertyfactory`?

no :cry: If you remove `context:property-placeholder` then it will crash. @groldan I think I already test `@environment` and it didn't work well but not sure, I'll test again. In any case, this stuff is more of a fix. We should fund a config refactor.

**Comment by severo** [comment]
Find or fund ? Both, I imagine

**Comment by RemiDesgrange** [comment]
@landryb The problem I'm facing right now, is that the properties file loader doesn't resolve var in properties file. Ex: 
```
publicurl=${schema}://{domainName}
```
display in html : `${schema}://{domainName}`

There are solutions but none of then worked ATM. I tried using `PropertySourcesPlaceholderConfigurer` [doc](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/context/support/PropertySourcesPlaceholderConfigurer.html) but it didn't work out.

**Comment by landryb** [comment]
i like the *make it work* commits, which clearly conveys the frustration of spending hours on an issue, and finally giving up for a "dammit, will be good enough" fix ;)

**Comment by severo** [comment]
Yes, I understand it the same way :) But, I think if you have spent various days on an issue, the commit message is a good occasion to give more details on what has changed, and maybe what has been tried, didn't work, why it was so hard, etc.

**Comment by RemiDesgrange** [comment]
Too bad that françois merged it. For this kind of stuff I like to squash before commit.  Gitlab has this usefull feature, and think this is great !

**Pull request merged**